src/metabase/upload.clj

```diff
@@ -590,19 +590,11 @@
       (throw (ex-info (tru "The CSV file contains duplicate column names.")
                       {:status-code 422})))
     (when (or extra missing)
-      (let [format-columns (fn [cols]
-                             (str/join ", " (map #(str "\"" % "\"") cols)))
-            error-message (cond
-                            (and extra missing)
-                            (tru "The CSV file contains extra columns that are not in the table: {0}. The CSV file is missing columns that are in the table: {1}."
-                                 (format-columns extra) (format-columns missing))
-                            extra
-                            (tru "The CSV file contains extra columns that are not in the table: {0}."
-                                 (format-columns extra))
-                            missing
-                            (tru "The CSV file is missing columns that are in the table: {0}."
-                                 (format-columns missing)))]
-        (throw (ex-info error-message {:status-code 422}))))))
+      (throw (ex-info "There were some error uploading your CSV file."
+                      (-> {:status-code 422
+                           ::schema-mismatch? true}
```

Comment:
Is using exception metadata like this for control flow a common and ordained pattern in the API?

Reply:
We don't have a common and ordained pattern. I'm open to suggestions for how to do this better.
See these issues for some discussion
https://github.com/metabase/metabase/issues/11175
https://github.com/metabase/metabase/issues/14683

Reply:
oh haha, I just mentioned #14683 in slack 😂